backend/icpy/agent/tools/youtube_download_tool.py

```
@@ -0,0 +1,505 @@
+                or context.get('cwd')
+                or os.environ.get('HOP_REMOTE_WORKSPACE_ROOT')
+                or posixpath.join('/home', remote_user, 'icotes')
+            )
+        
+        # Determine output directory
+        output_dir = Path(workspace_root) / self.DEFAULT_OUTPUT_DIR
+        
+        # Create output directory if it doesn't exist
+        try:
+            output_dir.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)